### PR TITLE
s/ECHConfigs/ECHConfigList, for clarity.

### PR DESCRIPTION
**Pull request by chris-wood**
See [the source of confusion](https://github.com/MikeBishop/dns-alt-svc/issues/298).

### CODE REVIEW

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -850,21 +850,21 @@ If the client offers the "ech_is_inner" extension ({{is-inner}})
 in addition to the "encrypted_client_hello" extension, the server MUST abort
 with an "illegal_parameter" alert.
 
-First, the server collects a set of candidate ECHConfigs. This set is
```
Comment (dmcardle) [inline draft-ietf-tls-esni.md:853]: Would it be possible to keep the "set of candidate $X" phrasing?

The phrase "candidate ECHConfigList" sounds to me like the whole ECHConfigList is a candidate, not that each element is a candidate. 

Perhaps

> First, the server collects a set of candidate ECHConfig values.

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -850,21 +850,21 @@ If the client offers the "ech_is_inner" extension ({{is-inner}})
...
-determined by one of the two following methods:
+First, the server collects a candidate ECHConfigList. This list is determined
```
Comment (davidben) [inline draft-ietf-tls-esni.md:853]: I think this section should just be "ECHConfigs" or "ECHConfig values". The server doesn't actually need to serialize them or anything.

Also now that we also have a better word, maybe {{real-ech}} should now be a bit more precise and say something like:
"To offer ECH, the client first chooses a suitable ECHConfig from the published ECHConfigList."

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (chris-wood) [reply draft-ietf-tls-esni.md:853]: Sure, that works equally well for me!

File: draft-ietf-tls-esni.md (hunk 2 above)
Comment (chris-wood) [reply draft-ietf-tls-esni.md]: (See latest change)

**Review by dmcardle — APPROVED** [review]
Thanks, Chris!

**Review by davidben — APPROVED** [review]
LGTM with one comment that got lost in the mixup. Now that we also have a better word, maybe {{real-ech}} should now be a bit more precise and say something like:

> To offer ECH, the client first chooses a suitable ECHConfig from the published ECHConfigList.

I figure the precision might help reenforce the path from published structure to client behavior.

**Comment by chris-wood** [comment]
> I figure the precision might help reenforce the path from published structure to client behavior.

I left this out because I figure there may be cases where there is no "published" ECHConfigList. (Maybe the config is baked in the app?)

**Comment by davidben** [comment]
Ah fair, that might be too precise. I think it still makes sense for that case to have the baked-in thing be an ECHConfigList than an ECHConfig, just for consistency. (I'm envisioning that ECHConfigList is the "public API" for client ECH and your TLS library might just that one byte string as input.)

How about s/published/server's/?

**Comment by chris-wood** [comment]
> How about s/published/server's/?

That'd work :)

**Comment by chris-wood** [comment]
@davidben done!

**Review by cbartle891 — APPROVED** [review]

**Review by davidben — APPROVED** [review]
LGTM

**Pull request merged**